compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/LangLibrary.java

```diff
@@ -0,0 +1,225 @@
+import org.wso2.ballerinalang.compiler.semantics.model.SymbolEnv;
+import org.wso2.ballerinalang.compiler.semantics.model.SymbolTable;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BInvokableSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BPackageSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.BSymbol;
+import org.wso2.ballerinalang.compiler.semantics.model.symbols.Symbols;
+import org.wso2.ballerinalang.compiler.util.CompilerContext;
+import org.wso2.ballerinalang.compiler.util.Name;
+import org.wso2.ballerinalang.compiler.util.Names;
+import org.wso2.ballerinalang.util.Flags;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.ballerina.compiler.api.symbols.TypeDescKind.ARRAY;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.BOOLEAN;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.DECIMAL;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.ERROR;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.FLOAT;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.FUTURE;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.INT;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.MAP;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.OBJECT;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.STREAM;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.STRING;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.TABLE;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.TYPEDESC;
+import static io.ballerina.compiler.api.symbols.TypeDescKind.XML;
+
+
+/**
+ * A class to hold the lang library function info required for types.
+ *
+ * @since 2.0.0
+ */
+public class LangLibrary {
+
+    private static final CompilerContext.Key<LangLibrary> LANG_LIB_KEY = new CompilerContext.Key<>();
+    private static final String LANG_VALUE = "value";
+
+    private final Map<String, Map<String, BInvokableSymbol>> langLibMethods;
+    private final Map<String, List<FunctionSymbol>> wrappedLangLibMethods;
+    private final SymbolFactory symbolFactory;
+
+    private LangLibrary(CompilerContext context) {
+        context.put(LANG_LIB_KEY, this);
+
+        symbolFactory = SymbolFactory.getInstance(context);
+        SymbolTable symbolTable = SymbolTable.getInstance(context);
+        Map<String, BPackageSymbol> langLibs = new HashMap<>();
+        wrappedLangLibMethods = new HashMap<>();
+
+        for (Map.Entry<BPackageSymbol, SymbolEnv> entry : symbolTable.pkgEnvMap.entrySet()) {
+            BPackageSymbol module = entry.getKey();
+            PackageID moduleID = module.pkgID;
+
+            if (Names.BALLERINA_ORG.equals(moduleID.orgName) &&
+                    (moduleID.nameComps.size() == 2 && Names.LANG.equals(moduleID.nameComps.get(0)))) {
```

Comment:
will the number of name components be 2 in future as well? What if we check for startsWith("lang.")?

Reply:
Good point. Don't think there's a restriction on that. Will change